x/oracle/keeper/oracle.go

```diff
@@ -42,6 +43,53 @@ func (k Keeper) RegisterOracle(ctx sdk.Context, msg *types.MsgRegisterOracle) er
 	return nil
 }
 
+func (k Keeper) UpdateOracleInfo(ctx sdk.Context, msg *types.MsgUpdateOracleInfo) error {
+	oracle, err := k.GetOracle(ctx, msg.OracleAddress)
+	if err != nil {
+		return sdkerrors.Wrapf(types.ErrUpdateOracle, err.Error())
+	}
+
+	blockTime := ctx.BlockHeader().Time
+
+	if !oracle.OracleCommissionRate.Equal(msg.OracleCommissionRate) {
+		if err := k.ValidateOracleCommission(oracle, blockTime, msg.OracleCommissionRate); err != nil {
+			return sdkerrors.Wrapf(types.ErrUpdateOracle, err.Error())
+		}
+	}
+
+	oracle.UpdateTime = blockTime
+	oracle.Endpoint = msg.Endpoint
+	oracle.OracleCommissionRate = msg.OracleCommissionRate
+
+	if err := k.SetOracle(ctx, oracle); err != nil {
+		return sdkerrors.Wrapf(types.ErrUpdateOracle, err.Error())
+	}
+	return nil
+}
+
+// ValidateOracleCommission validate an oracle's commission rate.
+// An error is returned if the new commission rate is invalid.
+func (k Keeper) ValidateOracleCommission(oracle *types.Oracle, blockTime time.Time, newRate sdk.Dec) error {
```

Comment:
It seems that this function doesn't have to be a keeper function.
Instead, how about to be a function of oracle? like,

```go
func (o Oracle) ValidateCommissionRate(blockTime time.Time, newRate sdk.Dec) error {
    // validate commission rate
}
```

Reply:
Thanks, there is no need for a keeper function because there is no store access. It's corrected.